app/src/org/commcare/models/database/SqlStorage.java

```diff
@@ -503,6 +505,10 @@ public void write(Persistable p) {
             update(p.getID(), p);
             return;
         }
+        int persistableSize = TableBuilder.toBlob(p).length;
```

Comment:
I think calling `toBlob` can be expensive because it re-serializes the entire object, which might get us into trouble with large cases / forms etc(?).
This change means that `toBlob` is being called twice, once in `DatabaseHelper.addDataToValues` and once here.
Can you think of a clean way to avoid this? 
Maybe passing in a `isSizeBounded` flag into the core code and only checking & throwing size exceptions on that? (definitely adds cruft to the code, but maybe okay in this situation..)